pkg/operator/controller/controller_lb.go

```diff
@@ -41,20 +40,37 @@ func (r *reconciler) ensureLoadBalancerService(ci *operatorv1.IngressController,
 	if err != nil {
 		return nil, err
 	}
-	if desiredLBService != nil && currentLBService == nil {
+
+	desired, exists := desiredLBService != nil, currentLBService != nil
+
+	switch {
+	case desired && !exists:
 		if err := r.client.Create(context.TODO(), desiredLBService); err != nil {
 			return nil, fmt.Errorf("failed to create load balancer service %s/%s: %v", desiredLBService.Namespace, desiredLBService.Name, err)
 		}
 		log.Info("created load balancer service", "namespace", desiredLBService.Namespace, "name", desiredLBService.Name)
 		return desiredLBService, nil
+	case !desired && exists:
+		err := r.finalizeLoadBalancerService(ci, dnsConfig)
```

Comment:
Is this resolved by making `finalizeLoadBalancerService` accept a service as input (eg. in this case we'd pass the current service rather than the desired)?

Reply:
`desiredDNSRecords` and thus `finalizeLoadBalancerService` are checking the endpoint publishing strategy type, which is an attribute of the ingresscontroller.  I suppose we could change `desiredDNSRecords` to take a service pointer and check whether it's nil as a proxy for checking whether the strategy type is LB service.